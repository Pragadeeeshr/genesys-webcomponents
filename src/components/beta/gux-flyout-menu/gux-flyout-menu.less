@import (reference) '../../../style/color.less';

// Style
gux-flyout-menu-beta {
  z-index: 2;
  color: @gux-type;

  .gux-flyout-menu-wrapper {
    display: none;
    flex-direction: column;
    border: 1px @gux-border;
    box-shadow: 0 0 2px fade(@gux-grey-global, 24%);
    background-color: @gux-type-dark;

    &.gux-shown {
      display: flex;
    }
  }

  .gux-arrow,
  .gux-arrow::after {
    position: absolute;
    content: ' ';
    height: 0;
    width: 0;
    border-style: solid;
    border-color: transparent;
  }

  .gux-arrow {
    border-width: 6px;
  }

  .gux-arrow::after {
    border-width: 4px;
  }

  &[data-popper-placement='bottom'],
  &[data-popper-placement='bottom-start'],
  &[data-popper-placement='bottom-end'] {
    margin-top: 6px;

    .gux-arrow {
      border-bottom-color: @gux-border;
      margin-top: -12px;
      margin-right: -12px;

      &::after {
        border-bottom-color: @gux-type-dark;
        margin-top: -2px;
        margin-left: -4px;
      }
    }
  }

  &[data-popper-placement='top'],
  &[data-popper-placement='top-start'],
  &[data-popper-placement='top-end'] {
    margin-bottom: 6px;

    .gux-arrow {
      bottom: 0;
      border-top-color: @gux-border;
      margin-bottom: -12px;

      &::after {
        border-top-color: @gux-type-dark;
        margin-top: -6px;
        margin-left: -4px;
      }
    }
  }

  &[data-popper-placement='left'],
  &[data-popper-placement='left-start'],
  &[data-popper-placement='left-end'] {
    margin-right: 6px;

    .gux-arrow {
      right: 0;
      border-left-color: @gux-border;
      margin-right: -12px;

      &::after {
        border-left-color: @gux-type-dark;
        margin-top: -4px;
        margin-left: -6px;
      }
    }
  }

  &[data-popper-placement='right'],
  &[data-popper-placement='right-start'],
  &[data-popper-placement='right-end'] {
    margin-left: 6px;

    .gux-arrow {
      border-right-color: @gux-border;
      margin-left: -12px;

      &::after {
        border-right-color: @gux-type-dark;
        margin-top: -4px;
        margin-left: -2px;
      }
    }
  }

  .gux-flyout-menu-content {
    margin: 0;
  }
}
