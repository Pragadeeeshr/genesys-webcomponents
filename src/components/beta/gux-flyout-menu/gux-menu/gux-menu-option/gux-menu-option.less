@import (reference) '../../../../../style/color.less';
@import (reference) '../gux-menu.common.less';

gux-menu-option {
  flex: 1 1 auto;
  align-self: auto;
  display: block;

  .gux-menu-option-button {
    all: unset;
    width: 100%;
    height: 32px;
    line-height: 32px;
    padding: 0 16px;
    background-color: @gux-off-white;
    color: @gux-type;
    width: @gux-menu-item-width;

    &:focus-within,
    &:hover {
      background-color: @gux-dark-blue;
      color: @gux-type-dark;
    }

    .gux-menu-option-button-text {
      display: block;
      white-space: nowrap;
      overflow-x: hidden;
      text-overflow: ellipsis;
    }
  }
}
