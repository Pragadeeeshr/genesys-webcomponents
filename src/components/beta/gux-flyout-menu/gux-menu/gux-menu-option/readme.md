# gux-flyout-menu



<!-- Auto Generated Below -->


## Methods

### `guxFocus() => Promise<void>`

Focus on the components button element

#### Returns

Type: `Promise<void>`




----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
