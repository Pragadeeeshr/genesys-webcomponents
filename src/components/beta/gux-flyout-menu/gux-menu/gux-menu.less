gux-menu {
  display: block;
  width: fit-content;
  overflow: hidden;
  padding: 8px 0;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-content: stretch;
  align-items: stretch;
}
