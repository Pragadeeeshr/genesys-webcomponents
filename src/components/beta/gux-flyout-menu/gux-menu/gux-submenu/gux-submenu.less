@import (reference) '../../../../../style/color.less';
@import (reference) '../gux-menu.common.less';

gux-submenu {
  flex: 1 1 auto;
  align-self: auto;
  display: block;

  .gux-submenu-button {
    all: unset;
    width: 100%;
    height: 32px;
    line-height: 32px;
    padding: 0 16px;
    background-color: @gux-off-white;
    color: @gux-type;
    width: @gux-menu-item-width;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: flex-start;
    align-content: stretch;
    align-items: center;

    &:focus-within,
    &:hover {
      background-color: @gux-dark-blue;
      color: @gux-type-dark;
    }

    .gux-submenu-button-text {
      order: 0;
      flex: 1 1 auto;
      align-self: auto;
      white-space: nowrap;
      overflow-x: hidden;
      text-overflow: ellipsis;
      margin-right: 12px;
    }

    .gux-submenu-open-icon {
      height: 10px;
      width: 10px;
      order: 0;
      flex: 0 0 auto;
      align-self: auto;
    }
  }

  .gux-submenu-wrapper {
    display: none;
    flex-direction: column;
    border: 1px @gux-border;
    box-shadow: 0 0 2px fade(@gux-grey-global, 24%);
    background-color: @gux-type-dark;
    padding: 8px 0;

    &.gux-shown {
      display: flex;
    }
  }
}
