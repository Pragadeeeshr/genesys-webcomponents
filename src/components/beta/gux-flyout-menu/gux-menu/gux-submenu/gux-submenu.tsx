import { createPopper, Instance } from '@popperjs/core';
import {
  Component,
  Element,
  h,
  Host,
  JSX,
  Listen,
  Method,
  Prop,
  State,
  Watch
} from '@stencil/core';

import {
  HTMLGuxMenuItemElement,
  hideDelay,
  menuNavigation
} from '../gux-menu.common';

@Component({
  styleUrl: 'gux-submenu.less',
  tag: 'gux-submenu'
})
export class GuxSubmenu {
  private hideDelayTimeout: NodeJS.Timer;
  private popperInstance: Instance;
  private buttonElement: HTMLButtonElement;
  private submenuElement: HTMLDivElement;
  private submenuContentElement: HTMLDivElement;

  @Element()
  private root: HTMLGuxSubmenuElement;

  @Prop()
  label: string;

  @State()
  private isShown: boolean = false;

  @Watch('isShown')
  forceUpdate(isShown: boolean) {
    if (isShown) {
      if (this.popperInstance) {
        this.popperInstance.update();
      }
    }
  }

  /**
   * Focus on the components button element
   */
  @Method()
  async guxFocus(): Promise<void> {
    this.buttonElement.focus();
  }

  @Listen('keydown')
  onKeydown(event: KeyboardEvent): void {
    menuNavigation(event, this.root);
    switch (event.key) {
      case 'ArrowRight':
      case 'Enter':
      case ' ':
        event.stopPropagation();

        this.show();
        this.focusOnSubmenu();
        break;

      case 'ArrowLeft':
      case 'Escape':
        if (this.isShown) {
          event.stopPropagation();
        }

        this.guxFocus();
        this.hide();
        break;
    }
  }

  @Listen('mouseenter')
  onmouseenter() {
    this.show();
  }

  @Listen('mouseleave')
  onMouseleave() {
    this.hide();
  }

  @Listen('click')
  onClick(event: MouseEvent) {
    if (this.submenuContentElement.contains(event.target as Node)) {
      return;
    }

    event.stopPropagation();
  }

  @Listen('focusin')
  onFocusin() {
    this.show();
  }

  @Listen('focusout')
  onFocusout() {
    this.hide();
  }

  private show(): void {
    clearTimeout(this.hideDelayTimeout);
    this.isShown = true;
  }

  private hide(): void {
    if (this.isShown) {
      this.hideDelayTimeout = setTimeout(() => {
        this.isShown = false;
      }, hideDelay);
    }
  }

  private runPopper(): void {
    this.popperInstance = createPopper(
      this.buttonElement,
      this.submenuElement,
      {
        modifiers: [
          {
            name: 'offset',
            options: {
              offset: [-8, 0]
            }
          },
          {
            name: 'flip',
            enabled: false
          }
        ],
        placement: 'right-start'
      }
    );
  }

  private destroyPopper(): void {
    if (this.popperInstance) {
      this.popperInstance.destroy();
      this.popperInstance = null;
    }
  }

  private focusOnSubmenu(): void {
    if (this.submenuContentElement.contains(document.activeElement)) {
      return;
    }

    const menuItems = Array.from(this.submenuContentElement.children);
    const nextFocusableElement = menuItems[0] as HTMLGuxMenuItemElement;

    nextFocusableElement.guxFocus();
  }

  componentDidLoad(): void {
    this.runPopper();
  }

  componentDidUnload(): void {
    this.destroyPopper();
  }

  render(): JSX.Element {
    return (
      <Host>
        <button
          type="button"
          class="gux-submenu-button"
          tabIndex={-1}
          ref={el => (this.buttonElement = el)}
          aria-haspopup="true"
          aria-expanded={this.isShown}
        >
          <span class="gux-submenu-button-text">{this.label}</span>
          <gux-icon
            class="gux-submenu-open-icon"
            icon-name="chevron-right"
            screenreader-text="Open submenu"
          ></gux-icon>
        </button>
        <div
          ref={el => (this.submenuElement = el)}
          class={{
            'gux-submenu-wrapper': true,
            'gux-shown': this.isShown
          }}
        >
          <div
            role="menu"
            class="gux-submenu-content"
            ref={el => (this.submenuContentElement = el)}
          >
            <slot />
          </div>
        </div>
      </Host>
    );
  }
}
