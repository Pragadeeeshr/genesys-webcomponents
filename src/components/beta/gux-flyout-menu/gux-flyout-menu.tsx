import { createPopper, Instance } from '@popperjs/core';
import {
  Component,
  Element,
  h,
  Host,
  JSX,
  Listen,
  State,
  Watch
} from '@stencil/core';

import { HTMLGuxMenuItemElement, hideDelay } from './gux-menu/gux-menu.common';

@Component({
  styleUrl: 'gux-flyout-menu.less',
  tag: 'gux-flyout-menu-beta'
})
export class GuxFlyoutMenu {
  private hideDelayTimeout: NodeJS.Timer;
  private popperInstance: Instance;
  private targetElement: HTMLSpanElement;
  private menuElement: HTMLDivElement;
  private menuContentElement: HTMLDivElement;

  @Element()
  private root: HTMLGuxMenuOptionElement;

  @State()
  private isShown: boolean = false;

  @Watch('isShown')
  forceUpdate(isShown: boolean) {
    if (isShown) {
      if (this.popperInstance) {
        this.popperInstance.update();
      }
    }
  }

  @Listen('keydown')
  onKeydown(event: KeyboardEvent): void {
    event.stopPropagation();

    if (this.isShown) {
      switch (event.key) {
        case 'Escape':
        case 'ArrowUp':
          this.hide();
          this.root.focus();
          return;

        case 'ArrowDown':
          this.focusOnMenu();
          return;
      }
    }

    switch (event.key) {
      case 'Enter':
      case 'ArrowDown':
      case ' ':
        this.show();
        return;
    }
  }

  @Listen('mouseenter')
  onmouseenter() {
    this.show();
  }

  @Listen('mouseleave')
  onMouseleave() {
    this.hide();
  }

  @Listen('click')
  onClick() {
    this.hide();
    this.root.focus();
  }

  @Listen('focusin')
  onFocusin() {
    this.show();
  }

  @Listen('focusout')
  onFocusout() {
    this.hide();
  }

  private show(): void {
    clearTimeout(this.hideDelayTimeout);
    this.isShown = true;
  }

  private hide(): void {
    if (this.isShown) {
      this.hideDelayTimeout = setTimeout(() => {
        this.isShown = false;
      }, hideDelay);
    }
  }

  private runPopper(): void {
    this.popperInstance = createPopper(this.targetElement, this.menuElement, {
      modifiers: [
        {
          name: 'offset',
          options: {
            offset: [0, 7]
          }
        },
        {
          name: 'arrow',
          options: {
            padding: 16 // 16px from the edges of the popper
          }
        }
      ],
      placement: 'bottom-start'
    });
  }

  private destroyPopper(): void {
    if (this.popperInstance) {
      this.popperInstance.destroy();
      this.popperInstance = null;
    }
  }

  private focusOnMenu(): void {
    if (this.menuContentElement.contains(document.activeElement)) {
      return;
    }

    const menu = this.menuContentElement.querySelector('gux-menu');
    const menuItems = Array.from(menu.children);
    const nextFocusableElement = menuItems[0] as HTMLGuxMenuItemElement;

    nextFocusableElement.guxFocus();
  }

  componentDidLoad(): void {
    this.runPopper();
  }

  componentDidUnload(): void {
    this.destroyPopper();
  }

  render(): JSX.Element {
    return (
      <Host tabIndex={1}>
        <span
          ref={el => (this.targetElement = el)}
          aria-haspopup="true"
          aria-expanded={this.isShown}
        >
          <slot name="target" />
        </span>
        <div
          ref={el => (this.menuElement = el)}
          class={{
            'gux-flyout-menu-wrapper': true,
            'gux-shown': this.isShown
          }}
        >
          <div
            class="gux-flyout-menu-content"
            ref={el => (this.menuContentElement = el)}
          >
            <slot name="menu" />
          </div>
          <div class="gux-arrow" data-popper-arrow />
        </div>
      </Host>
    );
  }
}
