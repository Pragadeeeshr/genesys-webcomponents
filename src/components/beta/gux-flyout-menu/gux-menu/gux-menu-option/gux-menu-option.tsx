import { Component, Element, h, JSX, Listen, Method } from '@stencil/core';

import { menuNavigation } from '../gux-menu.common';

@Component({
  styleUrl: 'gux-menu-option.less',
  tag: 'gux-menu-option'
})
export class GuxMenuOption {
  private buttonElement: HTMLButtonElement;

  @Element()
  private root: HTMLGuxMenuOptionElement;

  /**
   * Focus on the components button element
   */
  @Method()
  async guxFocus(): Promise<void> {
    this.buttonElement.focus();
  }

  @Listen('keydown')
  onKeydown(event: KeyboardEvent): void {
    menuNavigation(event, this.root);

    switch (event.key) {
      case 'ArrowRight':
        event.stopPropagation();

        break;
    }
  }

  render(): JSX.Element {
    return (
      <button
        type="button"
        class="gux-menu-option-button"
        role="menuitem"
        tabIndex={-1}
        ref={el => (this.buttonElement = el)}
      >
        <span class="gux-menu-option-button-text">
          <slot />
        </span>
      </button>
    );
  }
}
