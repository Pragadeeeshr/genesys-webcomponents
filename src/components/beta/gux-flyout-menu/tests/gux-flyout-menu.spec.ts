import { newSpecPage } from '@stencil/core/testing';
import '../../../../components';

import { GuxFlyoutMenu } from '../gux-flyout-menu';
import { GuxMenu } from '../gux-menu/gux-menu';
import { GuxSubmenu } from '../gux-menu/gux-submenu/gux-submenu';
import { GuxMenuOption } from '../gux-menu/gux-menu-option/gux-menu-option';

const components = [GuxFlyoutMenu, GuxMenu, GuxSubmenu, GuxMenuOption];
const html = `
  <gux-flyout-menu-beta>
    <span slot="target">Example Target Element</span>
    <gux-menu slot="menu">
      <gux-menu-option>Option One</gux-menu-option>
      <gux-submenu label="Submenu Two">
        <gux-menu-option>Option One</gux-menu-option>
        <gux-submenu label="Submenu Two">
          <gux-menu-option>Option One</gux-menu-option>
          <gux-menu-option>Option Two</gux-menu-option>
          <gux-menu-option>Option Three</gux-menu-option>
        </gux-submenu>
        <gux-menu-option>Option Three</gux-menu-option>
      </gux-submenu>
      <gux-menu-option>Option Three</gux-menu-option>
      <gux-menu-option>Option Four</gux-menu-option>
      <gux-submenu label="Submenu Five">
        <gux-menu-option>Option One</gux-menu-option>
        <gux-menu-option>Option Two</gux-menu-option>
        <gux-submenu label="Submenu Three">
          <gux-menu-option>Option One</gux-menu-option>
          <gux-submenu label="Submenu Two">
            <gux-menu-option>Option One</gux-menu-option>
            <gux-submenu label="Submenu Two">
              <gux-menu-option>Option One</gux-menu-option>
              <gux-menu-option>Option Two</gux-menu-option>
              <gux-menu-option>Option Three</gux-menu-option>
            </gux-submenu>
            <gux-menu-option>Option Three</gux-menu-option>
          </gux-submenu>
          <gux-menu-option>Option Three</gux-menu-option>
        </gux-submenu>
      </gux-submenu>
    </gux-menu>
  </gux-flyout-menu-beta>
`;
const language = 'en';

describe('gux-flyout-menu', () => {
  describe('#render', () => {
    it(`should render as expected`, async () => {
      const page = await newSpecPage({ components, html, language });

      expect(page.rootInstance).toBeInstanceOf(GuxFlyoutMenu);
      expect(page.root).toMatchSnapshot();
    });
  });

  // describe('#interactions', () => {
  //   it(`should change value on gux-switch-item click`, async () => {
  //     const page = await newSpecPage({ components, html, language });
  //     const element = page.root as HTMLGuxSwitchBetaElement;
  //     const guxSwitchItemMinute = page.root.querySelector(
  //       'gux-switch-item[value=minute]'
  //     ) as HTMLGuxSwitchItemElement;

  //     expect(element.value).toBe('day');

  //     guxSwitchItemMinute.click();
  //     await page.waitForChanges();

  //     expect(element.value).toBe(guxSwitchItemMinute.value);
  //   });

  //   it(`should not change value on gux-switch-item click if it is disabled`, async () => {
  //     const page = await newSpecPage({ components, html, language });
  //     const element = page.root as HTMLGuxSwitchBetaElement;
  //     const guxSwitchItemHour = page.root.querySelector(
  //       'gux-switch-item[value=hour]'
  //     ) as HTMLGuxSwitchItemElement;
  //     const currentValue = element.value;

  //     expect(currentValue).toBe('day');

  //     guxSwitchItemHour.click();
  //     await page.waitForChanges();

  //     expect(element.value).toBe(currentValue);
  //   });

  //   it(`should emit a 'guxvaluechanged' event when a new item is selected`, async () => {
  //     const page = await newSpecPage({ components, html, language });
  //     const element = page.root as HTMLGuxSwitchBetaElement;
  //     const guxSwitchItemMinute = page.root.querySelector(
  //       'gux-switch-item[value=minute]'
  //     ) as HTMLGuxSwitchItemElement;
  //     const eventSpy = jest.fn();

  //     element.addEventListener('guxvaluechanged', () => {
  //       eventSpy();
  //     });

  //     guxSwitchItemMinute.click();
  //     await page.waitForChanges();

  //     expect(eventSpy).toHaveBeenCalledWith();
  //   });

  //   it(`should not emit a 'guxvaluechanged' event when a disabled item is selected`, async () => {
  //     const page = await newSpecPage({ components, html, language });
  //     const element = page.root as HTMLGuxSwitchBetaElement;
  //     const guxSwitchItemHour = page.root.querySelector(
  //       'gux-switch-item[value=hour]'
  //     ) as HTMLGuxSwitchItemElement;
  //     const eventSpy = jest.fn();

  //     element.addEventListener('guxvaluechanged', () => {
  //       eventSpy();
  //     });

  //     guxSwitchItemHour.click();
  //     await page.waitForChanges();

  //     expect(eventSpy).not.toHaveBeenCalled();
  //   });
  // });
});
