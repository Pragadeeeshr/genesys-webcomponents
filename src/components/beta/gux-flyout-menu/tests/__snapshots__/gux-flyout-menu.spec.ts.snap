// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-flyout-menu #render should render as expected 1`] = `
<gux-flyout-menu-beta tabindex="1">
  <!---->
  <span aria-haspopup="true">
    <span slot="target">
      Example Target Element
    </span>
  </span>
  <div class="gux-flyout-menu-wrapper" style="position: absolute; left: 0; top: 0; margin: 0;">
    <div class="gux-flyout-menu-content">
      <gux-menu role="menu" slot="menu">
        <!---->
        <gux-menu-option>
          <!---->
          <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
            <span class="gux-menu-option-button-text">
              Option One
            </span>
          </button>
        </gux-menu-option>
        <gux-submenu label="Submenu Two">
          <!---->
          <button aria-haspopup="true" class="gux-submenu-button" tabindex="-1" type="button">
            <span class="gux-submenu-button-text">
              Submenu Two
            </span>
            <gux-icon class="gux-submenu-open-icon" icon-name="chevron-right" screenreader-text="Open submenu"></gux-icon>
          </button>
          <div class="gux-submenu-wrapper" style="position: absolute; left: 0; top: 0; margin: 0;">
            <div class="gux-submenu-content" role="menu">
              <gux-menu-option>
                <!---->
                <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                  <span class="gux-menu-option-button-text">
                    Option One
                  </span>
                </button>
              </gux-menu-option>
              <gux-submenu label="Submenu Two">
                <!---->
                <button aria-haspopup="true" class="gux-submenu-button" tabindex="-1" type="button">
                  <span class="gux-submenu-button-text">
                    Submenu Two
                  </span>
                  <gux-icon class="gux-submenu-open-icon" icon-name="chevron-right" screenreader-text="Open submenu"></gux-icon>
                </button>
                <div class="gux-submenu-wrapper" style="position: absolute; left: 0; top: 0; margin: 0;">
                  <div class="gux-submenu-content" role="menu">
                    <gux-menu-option>
                      <!---->
                      <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                        <span class="gux-menu-option-button-text">
                          Option One
                        </span>
                      </button>
                    </gux-menu-option>
                    <gux-menu-option>
                      <!---->
                      <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                        <span class="gux-menu-option-button-text">
                          Option Two
                        </span>
                      </button>
                    </gux-menu-option>
                    <gux-menu-option>
                      <!---->
                      <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                        <span class="gux-menu-option-button-text">
                          Option Three
                        </span>
                      </button>
                    </gux-menu-option>
                  </div>
                </div>
              </gux-submenu>
              <gux-menu-option>
                <!---->
                <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                  <span class="gux-menu-option-button-text">
                    Option Three
                  </span>
                </button>
              </gux-menu-option>
            </div>
          </div>
        </gux-submenu>
        <gux-menu-option>
          <!---->
          <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
            <span class="gux-menu-option-button-text">
              Option Three
            </span>
          </button>
        </gux-menu-option>
        <gux-menu-option>
          <!---->
          <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
            <span class="gux-menu-option-button-text">
              Option Four
            </span>
          </button>
        </gux-menu-option>
        <gux-submenu label="Submenu Five">
          <!---->
          <button aria-haspopup="true" class="gux-submenu-button" tabindex="-1" type="button">
            <span class="gux-submenu-button-text">
              Submenu Five
            </span>
            <gux-icon class="gux-submenu-open-icon" icon-name="chevron-right" screenreader-text="Open submenu"></gux-icon>
          </button>
          <div class="gux-submenu-wrapper" style="position: absolute; left: 0; top: 0; margin: 0;">
            <div class="gux-submenu-content" role="menu">
              <gux-menu-option>
                <!---->
                <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                  <span class="gux-menu-option-button-text">
                    Option One
                  </span>
                </button>
              </gux-menu-option>
              <gux-menu-option>
                <!---->
                <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                  <span class="gux-menu-option-button-text">
                    Option Two
                  </span>
                </button>
              </gux-menu-option>
              <gux-submenu label="Submenu Three">
                <!---->
                <button aria-haspopup="true" class="gux-submenu-button" tabindex="-1" type="button">
                  <span class="gux-submenu-button-text">
                    Submenu Three
                  </span>
                  <gux-icon class="gux-submenu-open-icon" icon-name="chevron-right" screenreader-text="Open submenu"></gux-icon>
                </button>
                <div class="gux-submenu-wrapper" style="position: absolute; left: 0; top: 0; margin: 0;">
                  <div class="gux-submenu-content" role="menu">
                    <gux-menu-option>
                      <!---->
                      <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                        <span class="gux-menu-option-button-text">
                          Option One
                        </span>
                      </button>
                    </gux-menu-option>
                    <gux-submenu label="Submenu Two">
                      <!---->
                      <button aria-haspopup="true" class="gux-submenu-button" tabindex="-1" type="button">
                        <span class="gux-submenu-button-text">
                          Submenu Two
                        </span>
                        <gux-icon class="gux-submenu-open-icon" icon-name="chevron-right" screenreader-text="Open submenu"></gux-icon>
                      </button>
                      <div class="gux-submenu-wrapper" style="position: absolute; left: 0; top: 0; margin: 0;">
                        <div class="gux-submenu-content" role="menu">
                          <gux-menu-option>
                            <!---->
                            <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                              <span class="gux-menu-option-button-text">
                                Option One
                              </span>
                            </button>
                          </gux-menu-option>
                          <gux-submenu label="Submenu Two">
                            <!---->
                            <button aria-haspopup="true" class="gux-submenu-button" tabindex="-1" type="button">
                              <span class="gux-submenu-button-text">
                                Submenu Two
                              </span>
                              <gux-icon class="gux-submenu-open-icon" icon-name="chevron-right" screenreader-text="Open submenu"></gux-icon>
                            </button>
                            <div class="gux-submenu-wrapper" style="position: absolute; left: 0; top: 0; margin: 0;">
                              <div class="gux-submenu-content" role="menu">
                                <gux-menu-option>
                                  <!---->
                                  <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                                    <span class="gux-menu-option-button-text">
                                      Option One
                                    </span>
                                  </button>
                                </gux-menu-option>
                                <gux-menu-option>
                                  <!---->
                                  <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                                    <span class="gux-menu-option-button-text">
                                      Option Two
                                    </span>
                                  </button>
                                </gux-menu-option>
                                <gux-menu-option>
                                  <!---->
                                  <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                                    <span class="gux-menu-option-button-text">
                                      Option Three
                                    </span>
                                  </button>
                                </gux-menu-option>
                              </div>
                            </div>
                          </gux-submenu>
                          <gux-menu-option>
                            <!---->
                            <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                              <span class="gux-menu-option-button-text">
                                Option Three
                              </span>
                            </button>
                          </gux-menu-option>
                        </div>
                      </div>
                    </gux-submenu>
                    <gux-menu-option>
                      <!---->
                      <button class="gux-menu-option-button" role="menuitem" tabindex="-1" type="button">
                        <span class="gux-menu-option-button-text">
                          Option Three
                        </span>
                      </button>
                    </gux-menu-option>
                  </div>
                </div>
              </gux-submenu>
            </div>
          </div>
        </gux-submenu>
      </gux-menu>
    </div>
    <div class="gux-arrow" data-popper-arrow style="position: absolute;"></div>
  </div>
</gux-flyout-menu-beta>
`;
